src/Foundation/NSAction.cs

```diff
@@ -55,27 +56,26 @@ public void Apply ()
 	}
 
 	// Use this for synchronous operations
-	[Register ("__MonoMac_ActionDispatcher")]
-	internal sealed class ActionDispatcher : NSObject {
+	[Register ("__MonoMac_NSActionDispatcher2")]
+	internal sealed class NSActionDispatcher2 : NSObject
+	{
 		public const string SelectorName = "xamarinApplySelector";
 		public static readonly Selector Selector = new Selector (SelectorName);
 
-		readonly Action action;
+		readonly SendOrPostCallback d;
+		readonly object state;
 
-		public ActionDispatcher (Action action)
+		public NSActionDispatcher2 (SendOrPostCallback d, object state)
 		{
-			if (action == null)
-				throw new ArgumentNullException ("action");
```

Comment:
I think the null check is still required

Reply:
Well, these calls come in from SynchronizationContext. If someone schedules a null task, there's bigger problems.

Reply:
While I agree with the reasoning of not having the null check, it is always a great idea to be defensive so :+1: on having the null check here